src/components/hooks/useDiagnostics.ts

```diff
@@ -0,0 +1,23 @@
+import { useContext } from 'react';
+
+import { FormtronComponentVariant } from '../../types';
+
+import { DiagnosticMessagesContext } from '../DiagnosticMessagesContext';
+
+export const useDiagnostics = (path: string[]) => {
+  const getMessages = useContext(DiagnosticMessagesContext);
+  const messages = getMessages(path);
+  let severity = -1;
+  let severityLabel = '';
+  for (const message of messages) {
+    if (message.severity > severity) {
+      severity = message.severity;
+      severityLabel = message.severityLabel;
+    }
+  }
+  let variant: FormtronComponentVariant = '';
+  if (severityLabel === 'warn') {
```

Comment:
Enum, please.

Reply:
OH MY GAWD I GET THE MESSAGE.

But just so you know... I think TS enums are evil and should be avoided. So if one day I say "I told you so" this is when I told ya.